clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -3665,6 +3670,52 @@ public void testWhenFetchResponseReturnsALeaderShipChangeErrorAndNewLeaderInform
         // Validate subscription is still valid & fetch-able for tp1.
         assertTrue(subscriptions.isFetchable(tp1));
     }
+    
+    @Test
+    public void testFetcherDontCacheAnyData() {
```

Comment:
This part is good but it's gone too far. Maybe we can create a `FetchResponse` and then test the method `responseData`?

Reply:
Yes, I will simplify this test